File: OLSource1/v1/VS_officedev/codesnippet/VisualBasic/how-to--add-chart-controls-to-worksheets_2.vb
    Private Sub AddChart()
        Dim NativeWorksheet As Microsoft.Office.Interop.Excel.Worksheet =
            Globals.ThisAddIn.Application.ActiveWorkbook.ActiveSheet

        Dim worksheet As Microsoft.Office.Tools.Excel.Worksheet =
            Globals.Factory.GetVstoObject(NativeWorksheet)

        Dim cells As Excel.Range = worksheet.Range("A5", "D8")
        Dim chart As Chart = worksheet.Controls.AddChart(cells, "employees")
        chart.ChartType = Excel.XlChartType.xl3DPie
        chart.SetSourceData(cells, Type.Missing)

    End Sub

File: OLSource1/v1/VS_officedev/codesnippet/CSharp/how-to--add-chart-controls-to-worksheets_2.cs
        private void AddChart()
        {
            Worksheet worksheet = Globals.Factory.GetVstoObject(
                Globals.ThisAddIn.Application.ActiveWorkbook.ActiveSheet);


            Excel.Range cells = worksheet.Range["A5", "D8"];
            Chart chart = worksheet.Controls.AddChart(cells, "employees");
            chart.ChartType = Microsoft.Office.Interop.Excel.XlChartType.xl3DPie;
            chart.SetSourceData(cells);
        }       